### PR TITLE
Added signup feature

### PR DESCRIPTION
**Pull request by jdx-code**
Sign up functionality for users.

### CODE REVIEW

**Review by bk7312 — CHANGES_REQUESTED** [review]
In `tsconfig.json`, please remove `src/lib/mongo/index.js` in line 24. All `.ts` files are already covered under `**/*.ts` and we're not supposed to include `.js` files.

Edit: Sorry, I don't think I was supposed to commit the requested change myself, need to learn how Raja did it.

File: event/src/app/api/users/signup/route.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            isSubscribed            
+        });
+
+        const savedUser: Users = await newUser.save();
+
+        return NextResponse.json({
+            message: "User created successfully",
+            success: true,
+            savedUser
+        });
```
Comment (jeisaRaja) [inline event/src/app/api/users/signup/route.ts]: As a best practice, we should avoid sending sensitive information like passwords or hashedPassword to the client.
I suggest modifying the response to exclude the password before sending it to the client. We can achieve this by omitting the password field from the serialized user object using the toJSON() method.

```suggestion
        const responseUser = savedUser.toJSON();
        delete responseUser.password;
        return NextResponse.json({
            message: "User created successfully",
            success: true,
            responseUser 
        });
```

**Review by jeisaRaja — CHANGES_REQUESTED** [review]
Looks good, but I have one suggestion about security measures

File: event/tsconfig.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "bundler",
+    "moduleResolution": "node",
```
Comment (bk7312) [inline event/tsconfig.json:10]: Consider changing `moduleResolution` back to the default `bundler`.

From the docs: `node10` (previously called `node`) for Node.js versions older than v10, which only support CommonJS require. You probably won’t need to use node10 in modern code.

https://www.typescriptlang.org/tsconfig#moduleResolution

**Review by bk7312 — CHANGES_REQUESTED** [review]

File: event/tsconfig.json (hunk 2 above)
Comment (bk7312) [inline event/tsconfig.json]: Consider changing `moduleResolution` back to the default `bundler`.

From the docs: `node10` (previously called `node`) for Node.js versions older than v10, which only support CommonJS `require`. You probably won’t need to use `node10` in modern code.

https://www.typescriptlang.org/tsconfig#moduleResolution

```suggestion
    "moduleResolution": "bundler",
```

**Review by bk7312 — CHANGES_REQUESTED** [review]

**Review by bk7312 — APPROVED** [review]

**Review by jeisaRaja — APPROVED** [review]

**Pull request merged**